packages/core/src/session.ts

```diff
@@ -283,14 +289,14 @@ class Session {
       connectionHolder,
       impersonatedUser: this._impersonatedUser,
       onClose: this._transactionClosed.bind(this),
-      onBookmarks: this._updateBookmarks.bind(this),
+      onBookmarks: (newBm, oldBm, db) => this._updateBookmarks(newBm, oldBm, db),
       onConnection: this._assertSessionIsOpen.bind(this),
       reactive: this._reactive,
       fetchSize: this._fetchSize,
       lowRecordWatermark: this._lowRecordWatermark,
       highRecordWatermark: this._highRecordWatermark
     })
-    tx._begin(this._lastBookmarks, txConfig)
+    tx._begin(() => this._bookmarks(), txConfig)
```

Comment:
Why do we deffer the bookmark fetching?

Reply:
Because we are only sure about the bookmarks after acquire the connection, it is a bit of leftover from the version where we needed to `manager.getAllBookmarks(['system', session.database])`

Reply:
Actually, now it is needed, because the begin method is not async